### PR TITLE
Rewrite access times in ISO8601 format

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/PkgServerLogAnalysis.jl [hunk 1]
```diff
@@ -155,7 +155,64 @@ function should_ignore_parse_failure(r::RegexMatch)
...
+            return nothing
+        end
+        dt === nothing && return nothing
+        offset = Hour(parse(Int, offset_hour)) + Minute(parse(Int, offset_min))
+        if offset_sign == "+"
+            dt -= offset
+        else # offset_sign == "-"
+            dt += offset
+        end
+        return Dates.format(dt, dateformat"yyyy-mm-ddTHH:MM:SS")
```
Comment (fredrikekre) [inline src/PkgServerLogAnalysis.jl:188]: Compared to before, this has the extra benefit that this column will be recognized as `Dates.DateTime` and not `String`.

**Comment by fredrikekre** [comment]
Given that all dates and times are scrambled, what is the point of including this data at all actually?

**Comment by staticfloat** [comment]
Can you give a little more explanation as to what this PR is trying to do?  All timestamps are already in UTC because the nginx processes themselves have their timezones set to UTC.

**Comment by fredrikekre** [comment]
Sorry, bad title; it is mostly about rewriting them into ISO8601 format. However, if they are all scrambled, why do we even include it in the datasets?

**Comment by staticfloat** [comment]
The scrambling is only done when anonymizing data, which we only do so that devs can have a test set to play with locally.  We don't anonymize the data as it's passing through the analysis passes.

File: src/PkgServerLogAnalysis.jl (hunk 1 above)
Comment (staticfloat) [reply src/PkgServerLogAnalysis.jl:188]: Yeah, this is nice.

**Comment by staticfloat** [comment]
With these changes are you able to run `bin/load_test_data.jl`?  We should probably `include()` that as part of the tests.

**Review by staticfloat — APPROVED** [review]

**Comment by fredrikekre** [comment]
> With these changes are you able to run bin/load_test_data.jl? We should probably include() that as part of the tests.

Yes, and done. Also added CI.

**Comment by fredrikekre** [comment]
<details>
<summary>Artifact download creating art.</summary>

```
Downloading artifact: anonymized_test_data
#=#=#                                                                         
##O#- #                                                                       
##O=#  #                                                                      
#=#=-#  #                                                                     
-#O#- #   #                                                                   
-=#=#   #   #                                                                 
-=O#- #  #    #                                                               
-=O=#  #   #   #                                                              
-=O=-#  #    #   #                                                            
-=O=- #   #   #     #                                                         
-=O=-   #   #   #     #                                                       
-=O=-    #    #    #    #                                                     
-=O=-      #   #     #    #                                                   
-=O=-        #   #     #     #                                                
-=O=-         #     #    #     #                                              
-=O=-           #     #    #     #                                            
-=O=-              #    #     #     #                                         
-=O=-                #    #     #     #                                       
-=O=-                  #     #     #     #                                    
-=O=-                    #     #     #     #                                  
-=O=-                      #     #     #      #                               
-=O=-                         #     #     #     #                             
-=O=-                           #     #     #     #                           
-=O=-                              #     #     #     #                        
-=O=-                                #     #     #     #                      
-=O=-                                  #      #    #     #                    
-=O=-                                     #     #     #    #                  
-=O=-                                       #     #     #    #                
-=O=-                                          #     #    #    #              
-=O=-                                            #     #    #    #            
-=O=-                                              #     #    #    #          
-=O=-                                                 #    #    #   #         
-=O=-                                                   #    #    #   #       
-=O=-                                                     #    #   #   #      
-=O=-                                                       #    #   #  #     
-=O=-                                                         #    #  #  #    
-=O=-                                                           #   #   # #   
-=O=-                                                             #   #  # #  
-=O=-                                                              #   #  # # 
-=O=-                                                                #  #  ## 
-=O=-                                                                 #  # # #
-=O=-                                                                   # # ##
-=O=-                                                                    # # #
-=O=-                                                                     # ##
-=O=-                                                                      ###
-=O=-                                                                      ###
-=O=-                                                                       ##
-=O=-                                                                      # #
-=O=-                                                                     # ##
-=O=-                                                                    # # #
-=O=-                                                                   #  ###
-=O=-                                                                  #  # ##
-=O=-                                                                 #  # # #
-=O=-                                                               #   # # # 
-=O=-                                                              #  #  # #  
-=O=-                                                            #   #  #  #  
-=O=-                                                          #   #   #  #   
-=O=-                                                        #    #   #  #    
-=O=-                                                      #    #   #   #     
-=O=-                                                    #    #    #  #       
-=O=-                                                  #    #    #   #        
-=O=-                                                #    #    #   #          
-=O=-                                             #     #    #    #           
-=O=-                                           #     #    #    #             
-=O=-                                         #    #     #    #               
-=O=-                                      #     #     #    #                 
-=O=-                                    #     #     #    #                   
-=O=-                                 #     #     #     #                     
-=O=-                               #     #     #     #                       
-=O=-                            #     #      #    #                          
-=O=-                          #     #     #     #                            
-=O=-                        #     #     #     #                              
-=O=-                     #     #     #     #                                 
-=O=-                   #     #     #     #                                   
-=O=-                 #    #     #     #                                      
-=O=-               #    #     #     #                                        
-=O=-            #     #     #     #                                          
-=O=-          #     #    #     #                                             
-=O=-         #    #    #     #                                               
-=O=-       #   #     #    #                                                  
-=O=-     #    #    #    #                                                    
-=O=-   #    #   #     #                                                      
-=O=-  #   #   #     #                                                        
-=O=- #  #    #    #                                                          
-=O=#   #   #   #                                                             
-=O#- #   #    #                                                              
-=#=-#  #    #                                                                
-#O=#  #   #                                                                  
-#O#- #  #                                                                    
#=#=#   #                                                                     
##O#- #                                                                       
#=#=-#                                                                        
##O=#                                                                         
##O#-                                                                         
#=#=-                                                                         
##O=-                                                                         
##O=-                                                                         
##O=-                                                                         
#=#=-                                                                         
##O#-                                                                         
#=#=#                                                                         
##O#- #                                                                       
##O=#  #                                                                      
#=#=-#  #                                                                     
-#O#- #   #                                                                   
-=#=#   #   #                                                                 
-=O#- #  #    #                                                               
-=O=#  #   #   #                                                              
-=O=-#  #    #   #                                                            
-=O=- #   #   #     #                                                         
-=O=-   #   #   #     #                                                       
-=O=-    #    #    #    #                                                     
-=O=-      #   #     #    #                                                   
-=O=-        #   #     #     #                                                
-=O=-         #     #    #     #                                              
-=O=-           #     #    #     #                                            
-=O=-              #    #     #     #                                         
-=O=-                #    #     #     #                                       
-=O=-                  #     #     #     #                                    
-=O=-                    #     #     #     #                                  
-=O=-                      #     #     #      #                               
-=O=-                         #     #     #     #                             
```

</details>

**Comment by staticfloat** [comment]
Thanks!

**Pull request merged**